src/legacy/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -405,8 +405,14 @@ function VisEditor(
     // update the searchSource when filters update
     const filterUpdateSubscription = subscribeWithScope($scope, queryFilter.getUpdates$(), {
       next: () => {
-        $scope.filters = queryFilter.getFilters();
-        $scope.fetch();
+        const filters = queryFilter.getFilters();
```

Comment:
I think I have caused this bug when I removed the [update \ fetch](https://github.com/elastic/kibana/blob/7.2/src/legacy/ui/public/filter_bar/query_filter.js#L384) optimization in filter manager.

Lets discuss whether this is a visualize specific problem, and then use this solution, or is it a more generic one, and then re-introduce this logic into filter manger.

Reply:
https://github.com/elastic/kibana/pull/41144 is related but we probably can't use the same way of propagating changes there because the embeddables have to know the actual current state of filters because they can also update them (see comment from  @stacey-gammon https://github.com/elastic/kibana/pull/41144#discussion_r303900315 ).